packages/docusaurus-plugin-content-blog/src/footnoteIDFixer.ts

```diff
@@ -0,0 +1,23 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import visit from 'unist-util-visit';
+import {simpleHash} from '@docusaurus/utils';
+import type {Transformer} from 'unified';
+import type {FootnoteReference, FootnoteDefinition} from 'mdast';
+
+export default function plugin(): Transformer {
+  return (root, vfile) => {
+    const suffix = `-${simpleHash(vfile.path!, 6)}`;
```

Comment:
Using the file path as the ID suffix may not be desirable, because if the file is moved, the anchor ID changes. However, (1) file movements should be rare and (2) nobody links to a footnote anchor anyways.